File: examples/julia-openai/test_transport.jl
using Test
include("transport.jl")

path = only(ARGS)
transport = StogasNative.Transport(path; configuration=Dict("environment"=>"staging", "security"=>"e2ee"))
@test startswith(transport.base_url, "http://127.0.0.1:")
close(transport)
close(transport)
@test transport.handle == C_NULL
@test_throws ErrorException StogasNative.Transport(path; configuration=Dict("security"=>"unknown"))
close(StogasNative.Transport(path; configuration=Dict("environment"=>"staging")))
println("native ownership and configuration passed")

File: examples/dart-openai/test_transport.dart
import 'transport.dart';

void main(List<String> args) {
  final library = args.single;
  final transport = StogasTransport(library, {
    'environment': 'staging',
    'security': 'e2ee',
  });
  if (!transport.baseUrl.startsWith('http://127.0.0.1:'))
    throw StateError('Missing local URL');
  transport.close();
  transport.close();
  var rejected = false;
  try {
    StogasTransport(library, {'security': 'unknown'}).close();
  } catch (_) {
    rejected = true;
  }
  if (!rejected) throw StateError('Invalid mode accepted');
  StogasTransport(library, {'environment': 'staging'}).close();
  print('native ownership and configuration passed');
}
